### DISCUSSION TITLE
How can i use date picker without using external css?

### DISCUSSION THREAD

**Discussion by protricker** [easepick/easepick#8] [Q&A] [answered]
Hello. your picker amazing. But i have a little problem. I use webpack and npm. I want to use date range plugin. But if i import css like 
```import '@easepick/bundle/dist/index.css';``` at the top of my js file I see only white screen instead of my page, something wrong with styles.
I found workaround how to use css in this plugin. But it is copy paste from original `@easepick/bundle/dist/index.css` to `build/date-range-picker.css`;

    const picker = new easepick.create({
        element: datePicker.get(0),
        css: [
            '/build/date-range-picker.css', // sad line :(
        ],
        repick: true,
        plugins: [RangePlugin],
        RangePlugin: {
            tooltipNumber(num) {
                return num - 1;
            },
            locale: {
                one: 'day',
                other: 'days',
            },
        },
    });

Could you please explain how can i use `css` prop and not use external link `https://cdn.jsdelivr.net/npm/@easepick/bundle@1.0.1/dist/index.css` as in documentation and not create copypast like `build/date-range-picker.css`

full code example:
```
import { easepick } from '@easepick/core';
import { RangePlugin } from '@easepick/range-plugin';
// import '@easepick/bundle/dist/index.css'; BREAKS page. 

jQuery(document).ready(function () {
    const datePicker = $('.js-datepicker');
    const picker = new easepick.create({
        element: datePicker.get(0),
        css: [
            '/build/date-range-picker.css', //Trying to import here 
        ],
        repick: true,
        plugins: [RangePlugin],
        RangePlugin: {
            tooltipNumber(num) {
                return num - 1;
            },
            locale: {
                one: 'day',
                other: 'days',
            },
        },
    });
});
```
Thank you.

**Comment by wakirin** [comment, marked as answer]
I see that you can use `raw-loader` ([stackoverflow](https://stackoverflow.com/questions/57460177/how-to-import-css-file-content-into-a-javascript-variable/57460784#57460784))

I tested a simple file and it works.
 index.ts:
```ts
import { easepick } from '@easepick/bundle';
import css_picker from '!!raw-loader!@easepick/bundle/dist/index.css';

const picker = new easepick.create({
  element: document.getElementById('datepicker'),
  css: css_picker,
});
```

I'd like to add a two remarks to the answer on SO:

1. I had to create `easepick.d.ts` in `./types` folder with this content:
```ts
declare module '*easepick/bundle/dist/index.css';
```
Otherwise I was getting an error: `TS2307: Cannot find module ...`

2. Instead of `import css_picker from 'raw-loader!@easepick/bundle/dist/index.css';` 
I use `import css_picker from '!!raw-loader!@easepick/bundle/dist/index.css';`
Otherwise, I didn't get the CSS file.
For your example you need to import CSS files from `core` and `range-plugin` package.
Also you need to add `types` folder in tsconfig.config:
```
{
  ...
  "include": [
    ...
    "types"
  ]
}
```

Perhaps some of these remarks may be solved with webpack config or TS config.
If you find another solution please share here.

**Reply by kl3sk** [reply]
Hello,

This technique works unless you use an array for `css`.

I manage to fix this with an extra file:

```js
// style.js

import css_picker from '!!raw-loader!@easepick/core/dist/index.css'
import css_picker_custom from '!!raw-loader!@styles/components/datepicker/custom.css'

export const styles = css_picker + css_picker_custom
```

and import this file like this:

```js
// picker.js

import {styles} from './styles'
new easepick.create({
     element,
     css: styles
 })
```

If there is a better way, please shre it.

**Reply by splittix** [reply]
I have the same issue. And I can't do an import since I'm using flat js files for a chrome extension vs. modules. Did you ever figure out how to do this with a local css file?

**Reply by yannickoo** [reply]
@kl3sk I couldn't imagine that the `+` syntax will work but that exactly was fixing the issue because the array is only for file paths but if you want to use multiple raw styles that is needed, thanks! 💪

**Reply by kl3sk** [reply]
Welcome. 

That's finally just a string concatenation 😉

**Reply by kl3sk** [reply]
@splittix you may try to create a local CSS file to import all in it. And call this file in easepick.

**Comment by beephotography** [comment]
Why do we even have to add CSS via configuration manually? For me, it doesn't make much sense. CSS should be included by default and be overwritable with custom CSS...
